photutils/psf/psfphotometry.py

```diff
@@ -0,0 +1,282 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+"""Module which provides classes to perform PSF Photometry"""
+
+from __future__ import division
+import abc, six
+import numpy as np
+from astropy.table import Table, vstack
+from photutils.psf import subtract_psf
+
+
+__all__ = ['PSFPhotometryBase', 'NStarPSFPhotometry']
+
+
+class PSFPhotometryBase(object):
+    __metaclass__ = abc.ABCMeta
+
+    @abc.abstractmethod
+    def do_photometry(self):
+        pass
+
+
+class NStarPSFPhotometry(PSFPhotometryBase):
+    """
```

Comment:
I'm pretty sure many people don't know it, e.g. I didn't  when I used
daophot as an undergrad, and only got the gotcha after used his variability
index, too for my master's thesis.

On 11 July 2016 at 16:58, Larry Bradley <notifications@github.com> wrote:

> In photutils/psf/psfphotometry.py
> <https://github.com/astropy/photutils/pull/385#discussion_r70285820>:
>
> > +from photutils.psf import subtract_psf
> > +
> > +
> > +__all__ = ['PSFPhotometryBase', 'NStarPSFPhotometry']
> > +
> > +
> > +class PSFPhotometryBase(object):
> > +    __metaclass__ = abc.ABCMeta
> > +
> > +    @abc.abstractmethod
> > +    def do_photometry(self):
> > +        pass
> > +
> > +
> > +class NStarPSFPhotometry(PSFPhotometryBase):
> > +    """
>
> I think I would have chosen "DAOPhot" or "DAOPhotPhotometry" since the
> name DAOPHOT is well known and the intent is clear. I'm not so sure
> everyone knows that Stetson wrote it.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/astropy/photutils/pull/385/files/ebaa7abb447e664e69674fb4ba56e15a9bf015a0#r70285820>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AGeUwo3TURqAMRV1pK10XKh8Ul69CjYVks5qUmg9gaJpZM4JDw9Q>
> .
>